### PR TITLE
View reliability

### PR DESCRIPTION
**Pull request by Emily-ejag**
Able to grab reliability and color not reliable to gray

### CODE REVIEW

**Comment by richford** [comment]
Task linked: [CU-868737x60 Visually encode run reliability in score reports](https://app.clickup.com/t/868737x60)

File: src/components/RoarDataTable.vue [hunk 1]
```diff
@@ -424,31 +424,59 @@ let toolTipByHeader = (header) => {
...
+  let index = -1;
   if (colHeader === 'Phoneme' && colData.scores?.pa?.standard) {
     toolTip += colData.scores.pa?.support_level + '\n' + '\n';
     toolTip += 'Percentile: ' + colData.scores?.pa?.percentile + '\n';
     toolTip += 'Raw Score: ' + colData.scores?.pa?.raw + '\n';
     toolTip += 'Standardized Score: ' + colData.scores?.pa?.standard + '\n';
+    index = getIndexTask(colData, 'pa');
+    if(colData.assignment.assessments[index].engagementFlags !== undefined && colData.assignment.assessments[index].engagementFlags !== '' && !colData.assignment.assessments[index].reliable){
+      toolTip += '\n' + 'Reliability: ' + Object.keys(colData.assignment.assessments[index].engagementFlags) + '\n';
+    }
```
Comment (richford) [inline src/components/RoarDataTable.vue:446]: I'd suggest here:
- Saving the flags to a variable for readability
- Converting the camel cased flags to lower case and separating by commas and spaces

```suggestion
    const flags = colData.assignment.assessments[index].engagementFlags;
    if(flags !== undefined && flags !== '' && !colData.assignment.assessments[index].reliable){
      toolTip += '\n' + 'Reliability: ' + Object.keys(flags).map(_lowerCase).join(', ') + '\n';
    }
```

This suggestion uses [lodash's `lowerCase` method](https://lodash.com/docs/4.17.15#lowerCase), so you'll have to import that above as well.

File: src/components/RoarDataTable.vue [hunk 2]
```diff
@@ -424,31 +424,59 @@ let toolTipByHeader = (header) => {
...
   } else if (colHeader === 'Word' && colData.scores?.swr?.standard) {
     toolTip += colData.scores?.swr?.support_level + '\n' + '\n';
     toolTip += 'Percentile: ' + colData.scores?.swr?.percentile + '\n';
     toolTip += 'Raw Score: ' + colData.scores?.swr?.raw + '\n';
     toolTip += 'Standardized Score: ' + colData.scores?.swr?.standard + '\n';
+    index = getIndexTask(colData, 'swr');
+    if(colData.assignment.assessments[index].engagementFlags !== undefined && colData.assignment.assessments[index].engagementFlags !== '' && !colData.assignment.assessments[index].reliable){
+      toolTip += '\n' + 'Reliability: ' + Object.keys(colData.assignment.assessments[index].engagementFlags) + '\n';
+    }
```
Comment (richford) [inline src/components/RoarDataTable.vue:455]: See above suggestion for PA.

File: src/components/RoarDataTable.vue [hunk 3]
```diff
@@ -424,31 +424,59 @@ let toolTipByHeader = (header) => {
...
   } else if (colHeader === 'Sentence' && colData.scores?.sre?.standard) {
     toolTip += colData.scores?.sre?.support_level + '\n' + '\n';
     toolTip += 'Percentile: ' + colData.scores?.sre?.percentile + '\n';
     toolTip += 'Raw Score: ' + colData.scores?.sre?.raw + '\n';
     toolTip += 'Standardized Score: ' + colData.scores?.sre?.standard + '\n';
+    index = getIndexTask(colData, 'sre');
+    if(colData.assignment.assessments[index].engagementFlags !== undefined && colData.assignment.assessments[index].engagementFlags !== '' && !colData.assignment.assessments[index].reliable){
+      toolTip += '\n' + 'Reliability: ' + Object.keys(colData.assignment.assessments[index].engagementFlags) + '\n';
+    }
```
Comment (richford) [inline src/components/RoarDataTable.vue:464]: See above suggestion for PA

File: src/components/RoarDataTable.vue [hunk 4]
```diff
@@ -424,31 +424,59 @@ let toolTipByHeader = (header) => {
...
   } else if (colHeader === 'Letter Names and Sounds' && colData.scores?.letter) {
     toolTip += 'Raw Score: ' + colData.scores?.letter?.raw + '\n';
+    index = getIndexTask(colData, 'letter');
+    if(colData.assignment.assessments[index].engagementFlags !== undefined && colData.assignment.assessments[index].engagementFlags !== '' && !colData.assignment.assessments[index].reliable){
+      toolTip += '\n' + 'Reliability: ' + Object.keys(colData.assignment.assessments[index].engagementFlags) + '\n';
+    }
```
Comment (richford) [inline src/components/RoarDataTable.vue:470]: See above suggestion for PA

File: src/components/RoarDataTable.vue [hunk 5]
```diff
@@ -424,31 +424,59 @@ let toolTipByHeader = (header) => {
...
   } else if (colHeader === 'Palabra' && colData.scores?.['swr-es']?.standard) {
     toolTip += colData.scores?.['swr-es'].support_level + '\n' + '\n';
     toolTip += 'Percentile: ' + colData.scores?.['swr-es']?.percentile + '\n';
     toolTip += 'Raw Score: ' + colData.scores?.['swr-es']?.raw + '\n';
     toolTip += 'Standardized Score: ' + colData.scores?.['swr-es']?.standard + '\n';
+    index = getIndexTask(colData, 'swr-es');
+    if(colData.assignment.assessments[index].engagementFlags !== undefined && colData.assignment.assessments[index].engagementFlags !== '' && !colData.assignment.assessments[index].reliable){
+      toolTip += '\n' + 'Reliability: ' + Object.keys(colData.assignment.assessments[index].engagementFlags) + '\n';
+    }
```
Comment (richford) [inline src/components/RoarDataTable.vue:479]: See above suggestion for PA.

**Review by richford — CHANGES_REQUESTED** [review]
Thanks @Emily-ejag. I left some in-line comments with suggestions for improvements. But a larger issue is that this breaks filtering on scores.

**Comment by richford** [comment]
I'm also wondering about the visual encoding used here. Just my two cents here but I'd rather keep the colors of the circles consistent and shade the "background" of the table cell or something like that.

**Comment by Emily-ejag** [comment]
Created a function to print the flags, since it was too much code :) , waiting for filtering to get fixed

**Review by richford — CHANGES_REQUESTED** [review]

**Comment by Emily-ejag** [comment]
> This is looking great! @Emily-ejag could you please resolve the merge conflict? After that I think we are good to go!

Thank you!, Will do, I also want to add the text Carrie gives me first :)

File: src/components/RoarDataTable.vue [hunk 6]
```diff
@@ -436,42 +437,57 @@ function getIndexTask(colData, task){
...
   if(flags !== undefined && flags !== '' && !ColData.assignment.assessments[index].reliable){
-    return '\n' + 'Reliability: ' + Object.keys(flags).map(_lowerCase).join(', ') + '\n';
+    const reliabilityFlags = Object.keys(flags).map(flag => {
+      switch (flag) {
+        case 'notEnoughResponses':
+          return 'Assessment was incomplete';
+        case 'responseTimeTooFast':
+          return 'Responses were too fast';
+        default:
+          return flag;
```
Comment (richford) [inline src/components/RoarDataTable.vue:447]:
```suggestion
          return _lowerCase(flag);
```

File: src/components/RoarDataTable.vue [hunk 7]
```diff
@@ -249,8 +249,8 @@ import _find from 'lodash/find';
 import _filter from 'lodash/filter';
 import _toUpper from 'lodash/toUpper';
 import _startCase from 'lodash/startCase';
-import _lowerCase from 'lodash/lowerCase';
-import { taskFilterBlacklist } from '@/helpers/reports';
+// import _lowerCase from 'lodash/lowerCase';
```
Comment (richford) [inline src/components/RoarDataTable.vue:252]: Maybe we can keep this in here to handle the fallback case in the switch statement below
```suggestion
import _lowerCase from 'lodash/lowerCase';
```

File: src/components/RoarDataTable.vue [hunk 8]
```diff
@@ -436,42 +437,57 @@ function getIndexTask(colData, task){
...
+      switch (flag) {
+        case 'notEnoughResponses':
+          return 'Assessment was incomplete';
+        case 'responseTimeTooFast':
+          return 'Responses were too fast';
+        default:
+          return flag;
+      }
+    });
+    // return '\n' + 'Reliability: ' + Object.keys(flags).map(_lowerCase).join(', ') + '\n';
```
Comment (richford) [inline src/components/RoarDataTable.vue:450]:
```suggestion
```

**Review by richford — CHANGES_REQUESTED** [review]
@Emily-ejag, I requested some very minor adjustments. But then I think we are good to merge. Could you do `npm run format` also before pushing up the changes? Thanks for all of your hard work on this!

**Review by richford — APPROVED** [review]

**Pull request merged**